server/config_manager/config_manager.go

```diff
@@ -486,51 +493,115 @@ func (lc *LocalConfig) getConfigs() map[string]interface{} {
 	}
 
-	t := reflect.TypeOf(res[configName[0]])
-	if t == nil {
-		return errors.Errorf("config item %s is not existed", configName[0])
+	_, ok := config[configName[0]]
+	if !ok {
+		// TODO: remove it
+		if configName[0] != "schedulers-v2" {
+			return errors.Errorf("cannot find the config item: %v", configName[0])
+		}
 	}
-	// TODO: support more types
-	var v interface{}
-	var err error
-	switch t.Kind() {
-	case reflect.Bool:
-		v, err = strconv.ParseBool(value)
-	case reflect.Int:
-		v, err = strconv.Atoi(value)
-	case reflect.Int64:
-		v, err = strconv.ParseInt(value, 10, 64)
-	case reflect.Float64:
-		v, err = strconv.ParseFloat(value, 64)
-	case reflect.String:
-		v = value
-	case reflect.Slice:
-		// TODO: make slice work for any other type
-		v = strings.Split(value, ",")
-	default:
-		return errors.Errorf("unsupported type")
+
+	container := make(map[string]interface{})
+
+	// TODO: remove it
+	if configName[0] == "cluster-version" {
+		cv, err := cluster.ParseVersion(value)
+		if err != nil {
+			return errors.Errorf("failed to parse version: %v", err.Error())
+		}
+		container[configName[0]] = cv
+	} else if _, err := toml.Decode(value, &container); err != nil {
+		if !strings.Contains(err.Error(), "bare keys") {
+			return errors.Errorf("failed to decode value: %v", err.Error())
+		}
+		container[configName[0]] = value
+	} else if configName[0] == "label-property" {
+		config[configName[0]] = container
+		return nil
 	}
 
+	v, err := getUpdateValue(config[configName[0]], container[configName[0]])
 	if err != nil {
 		return err
 	}
-	res[configName[0]] = v
 
+	config[configName[0]] = v
 	return nil
 }
 
+func getUpdateValue(item, updateItem interface{}) (interface{}, error) {
+	var err error
+	var v interface{}
+	var tmp float64
+	switch item.(type) {
+	case bool:
+		switch t1 := updateItem.(type) {
+		case string:
+			v, err = strconv.ParseBool(updateItem.(string))
+		case bool:
+			v = updateItem
+		default:
+			return nil, errors.Errorf("unexpected type: %T\n", t1)
+		}
+	case int64:
+		switch t1 := updateItem.(type) {
+		case string:
+			tmp, err = strconv.ParseFloat(updateItem.(string), 64)
+			v = int64(tmp)
+		case float64:
+			v = int64(updateItem.(float64))
+		case int64:
+			v = updateItem
+		default:
+			return nil, errors.Errorf("unexpected type: %T\n", t1)
+		}
+	case []interface{}:
+		strSlice := strings.Split(updateItem.(string), ",")
+		slice := make([]interface{}, 0)
+		for _, str := range strSlice {
+			slice = append(slice, str)
+		}
+		v = slice
+	case float64:
+		switch t1 := updateItem.(type) {
+		case string:
+			v, err = strconv.ParseFloat(updateItem.(string), 64)
+		case float64:
+			v = updateItem
+		default:
+			return nil, errors.Errorf("unexpected type: %T\n", t1)
+		}
+	case string:
+		switch t1 := updateItem.(type) {
+		case string:
+			v = updateItem
+		default:
+			return nil, errors.Errorf("unexpected type: %T\n", t1)
+		}
+	case nil:
+	default:
+		// TODO: support more types
+		v = updateItem
```

Comment:
How about returning an error? I think we should be strict about unknown types in case of unexpected errors.